lib/mastodon/snowflake.rb

```diff
@@ -145,6 +145,12 @@ def id_at(timestamp)
       id
     end
 
+    def id_at_start(timestamp)
+      id = timestamp.to_i * 1000
+      id = id << 16
+      id
```

Comment:
I would maybe one-line the whole thing like `(timestamp.to_i * 1000) << 16`

Reply:
But *maybe* it would be better to modify `id_at` to take an option for the `rand` part?